### PR TITLE
332-list-triggers: list all triggers

### PR DESCRIPTION
**Pull request by abdul-shajin**
Added model, migrations
Added GraphQL Query
Added GraphQL Tests
Added Model tests

Issue #332

### CODE REVIEW

**Comment by tejaswinichile** [comment]
@abdul-shajin _a test cases are failing, can you have a look.

**Comment by abdul-shajin** [comment]
@tejaswinichile _a some tests in resolvers were failing. Fixed

File: db/migrate/20180807160031_create_triggers.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+class CreateTriggers < ActiveRecord::Migration[5.2]
+  def change
+    create_table :triggers, id: :uuid do |t|
+      t.string :slug, null: false
```
Comment (tejaswinichile) [inline db/migrate/20180807160031_create_triggers.rb]: I think this should be unique. I don't know exact requirement here. Can you confirm this on the issue?

**Comment by tejaswinichile** [comment]
@abdul-shajin  _a

File: db/migrate/20180807160031_create_triggers.rb (hunk 1 above)
Comment (neerajdotname) [reply db/migrate/20180807160031_create_triggers.rb]: Why do we need a slug for a trigger?

**Comment by abdul-shajin** [comment]
@tejaswinichile  _a  I added unique constraint and the index

**Comment by neerajdotname** [comment]
@abdul-shajin Please do not forget to end sentence with a period. Also in that sentence you missed out "have".

It should have been like this.

> I have added unique constraint and the index.

File: app/models/trigger.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class Trigger < ApplicationRecord
+
+  scope :all_active, -> { where(active: true) }
```
Comment (tejaswinichile) [inline app/models/trigger.rb]: Just `active` will be sufficient, instead of `all_active`. Same for `inactive`.

**Comment by abdul-shajin** [comment]
Thanks Neeraj. Will keep that in mind..

**Comment by abdul-shajin** [comment]
@tejaswinichile _a Done the review suggestions.

**Pull request merged**